### ISSUE TITLE
Having different order of bits in XML data fails to parse

### ISSUE THREAD

**Issue by abhikeshav** [CESNET/libyang#175]
For below yang:

```
 leaf bits-value {
          description "this is bits type value";
          type bits {
            bit disable-nagle {
              position 0;
            }
            bit auto-sense-speed {
              position 1;
            }
          }
          default "auto-sense-speed";
        }
```

Both below XML data fails with error:
**1)**

```
     <bits-value>auto-sense-speed disable-nagle</bits-value>
```

```
libyang[0]: Invalid value "auto-sense-speed disable-nagle" in "bits-value" element. (path: /a:x/bits-value)
```

**2)**

```
<bits-value>auto-sense-speed </bits-value>
```

```
libyang[0]: Invalid value "auto-sense-speed " in "bits-value" element. (path: /a:x/bits-value)
```

But having bits in same order is successfully parsed:

```
     <bits-value>disable-nagle auto-sense-speed</bits-value>
```

**Comment by rkrejci** [comment]
Well, it follows the canonical form - RFC 7950 (as well as 6020 ) in 9.7.3 says:
```
In the canonical form, the bit values are separated by a single space
character and they appear ordered by their position.
```

It is true, that it should rather be able to accept lexical representation of the bits and internally convert it to the canonical representation. Fixing this issue may take some time.

**Comment by abhikeshav** [comment]
Thanks for the update and for fixing the other issues. Much appreciated!

Thanks,
Abhi

**Comment by rkrejci** [comment]
it should be fixed in devel (0.11.33)
